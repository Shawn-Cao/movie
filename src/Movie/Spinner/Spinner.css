.Spinner {
  animation: Spinner-spin infinite 3s linear;
  height: 100px;
  width: 100px;
  margin-left: -50px;
  margin-top: -50px;
}

@keyframes Spinner-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
