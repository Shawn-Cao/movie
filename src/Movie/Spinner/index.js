import React, { Component } from 'react';
import PropTypes from 'prop-types';
import logo from '../../logo.svg';
import './Spinner.css';

class Spinner extends Component {
  static get propTypes() {
    return {
      className: PropTypes.string,
      style: PropTypes.object
    };
  }

  render() {
    return (
      <div className={this.props.className} style={this.props.style}>
        <img src={logo} className="Spinner" alt="loading spinner" />
      </div>
    );
  }
}

export default Spinner;
