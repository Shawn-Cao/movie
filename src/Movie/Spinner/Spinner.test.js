import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Spinner from './index';

describe('MovieTime', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Spinner />, div);
  });
  it('matches snapshot when spinning', () =>{
    const tree = renderer.create(
      <Spinner spin={true} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('matchss snapshot when not spinning', () =>{
    const tree = renderer.create(
      <Spinner spin={false} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
