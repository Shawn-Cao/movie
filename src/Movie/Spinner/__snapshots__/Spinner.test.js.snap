// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTime matches snapshot when spinning 1`] = `
<div
  className={undefined}
  style={undefined}
>
  <img
    alt="loading spinner"
    className="Spinner"
    src="logo.svg"
  />
</div>
`;

exports[`MovieTime matchss snapshot when not spinning 1`] = `
<div
  className={undefined}
  style={undefined}
>
  <img
    alt="loading spinner"
    className="Spinner"
    src="logo.svg"
  />
</div>
`;
